tools/generator/templates/includes/Makefile.language.jinja2

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2020 Association Prologin <association@prologin.org>
+
+# Add all your champion files in here, separated by spaces
+# Include all the sources, headers and README files.
+CHAMPION_FILES = {% block champion_files %}{% endblock %}
+
+# ----
+# Do not modify the lines below
+
+CHAMPION_FILES += {% block generated_files %}interface.cc{% endblock %}
+
+CPPFLAGS += -MMD -MP
+CXXFLAGS = -ggdb3 -Wall -std=c++17 -fPIC
+{% block extra_flags %}
+{% endblock %}
+
+to_clean :=
+
+cxx_sources = $(filter %.cc,$(CHAMPION_FILES))
+objs += $(cxx_sources:.cc=.o)
+
+{% block extra_rules %}
+{% endblock %}
+
+deps = $(objs:.o=.d)
```

Comment:
I don’t think every .o obtained by any language has a corresponding .d file.

I don’t like that the dependency is done on the main makefile. the -MMD flags is specific to C/C++, and should be present for other languages. Each language will have to manage their own dependencies anyway.